mitiq/interface/mitiq_qibo/conversions.py

```diff
@@ -0,0 +1,529 @@
+                    "Qubit {} is not defined in QASM code." "".format(qubit),
+                )
+
+            register, idx = next(read_args(args_list[1]))
+            if register not in cregs_size:
+                raise_error(
+                    ValueError,
+                    "Classical register name {} is not defined "
+                    "in QASM code.".format(register),
+                )
+            if idx >= cregs_size[register]:
+                raise_error(
+                    ValueError,
+                    "Cannot access index {} of register {} "
+                    "with {} qubits."
+                    "".format(idx, register, cregs_size[register]),
+                )
+            if register in registers:
+                registers_not_none = {key: value for key, value in registers.items() if value is not None}
+                if idx in registers_not_none[register]:
+                    raise_error(
+                        KeyError,
+                        "Key {} of register {} has already "
+                        "been used.".format(idx, register),
+                    )
+                if registers[register] is not None:
+                    registers[register][idx] = qubits[qubit]
+            else:
+                registers[register] = {idx: qubits[qubit]}
+                gate_list.append(("M", register, None))
+
+        else:
+            pieces = [x for x in re.split("[()]", command) if x]
+            if len(pieces) == 1:
+                gatename, params = pieces[0], None
+                if gatename not in gates.QASM_GATES:
+                    if gatename != "sx" and gatename != "sxdg": 
+                        raise_error(
+                            ValueError,
+                            "QASM command {} is not recognized." "".format(command),
+                        )
+                if gatename in gates.PARAMETRIZED_GATES:
+                    raise_error(
+                        ValueError,
+                        "Missing parameters for QASM " "gate {}.".format(gatename),
+                    )
+
+            elif len(pieces) == 2:
+                gatename = pieces[0]
+                params = pieces[1].replace(" ", "").split(",")
+                if gatename not in gates.PARAMETRIZED_GATES:
+                    raise_error(
+                        ValueError, "Invalid QASM command {}." "".format(command)
+                    )
+                try:
+                    for i, p in enumerate(params):
+                        denominator = 1
+                        if "/" in p:
+                            p, denominator = p.split("/")
+                        if "pi" in p:
+                            from functools import reduce
+                            from operator import mul
+
+                            s = p.replace("pi", str(pi)).split("*")
+                            p = reduce(mul, [float(j) for j in s], 1)
+                        params[i] = float(p)
```

Comment:
Params is initally a list of strings mainly numbers, for example params=['0']. 
In the end of the function we append the params to gate list*, but this params should be a list of floats, params=[ 0 ]. 
In the case of having a pi in params we have initially params=['pi*-0.5']
After the conversion (entering an if to deal with 'pi') we have params=[1.5707963267948966] where the number is type float. 
If this is causing problems to mypy because of the change of type in the list we could define another list of floats and add to this list the floats without changing the types in the initial params list. 
*Line where we append params to gate list
gate_list.append((gates.QASM_GATES[gatename], list(qubit_list), params))

Reply:
`gate_list.append((gates.QASM_GATES[gatename], list(qubit_list), params))`
This line was causing an error because, as you pointed out, params was initially set to a list of strings but converted into a list of floats mid-code. To solve the error mypy was rasing I've implemented the following: 
`float_params = None if params is None else [float(element) for element in params]
gate_list.append((str(gates.QASM_GATES[gatename]), list(qubit_list), float_params))`